R/processes.R

```diff
@@ -96,7 +98,8 @@ create_processes <- function(
     targeted_process = function(timestep, target){
       infection_outcome_process(timestep, target, 
                                 variables, renderer, parameters, 
-                                prob = rate_to_prob(infection_outcome$rates))
+                                prob = rate_to_prob(infection_outcome$rates),
+                                relative_rates = infection_outcome$relative_rates)
```

Comment:
I've only just realised how ugly and genius this is...

I don't yet know how I feel about it

Reply:
I think the above edit to CompetingOutcome is a better way to deal with optional arguments for the processes. That way it generalises to different competing hazard scenarios and you don't have to continue adding fields to the R6 object for more arguments.